File: articles/decision-hygiene-and-bias-reduction/r/run_all_decision_hygiene_workflows.R
# run_all_decision_hygiene_workflows.R
# Run all R workflows for Decision Hygiene and Bias Reduction.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_hygiene_bias_reduction_workflow.R",
  "bias_noise_summary_tables.R",
  "calibration_review_tables.R",
  "hygiene_practice_reports.R",
  "evidence_quality_review.R",
  "decision_hygiene_review_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision hygiene workflows completed.")

File: articles/decision-hygiene-and-bias-reduction/python/run_all_decision_hygiene_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Hygiene and Bias Reduction."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "bias_noise_decomposition.py",
    "calibration_review.py",
    "evidence_quality_checks.py",
    "framing_bias_checks.py",
    "structured_dissent_review.py",
    "decision_hygiene_review_queue.py",
    "decision_record_exporter.py",
    "decision_hygiene_bias_reduction_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision hygiene workflows completed.")


if __name__ == "__main__":
    main()
